lib/customer-deposit-wallet-pure/haskell/Cardano/Wallet/Deposit/Pure/RollbackWindow.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.Deposit.Pure.RollbackWindow where
+
+if' :: Bool -> a -> a -> a
+if' True thn els = thn
+if' False thn els = els
+
+data RollbackWindow time = RollbackWindowC
+    { finality :: time
+    , tip :: time
+    }
+
+member :: Ord time => time -> RollbackWindow time -> Bool
+member time w = finality w <= time && time <= tip w
+
+singleton :: Ord time => time -> RollbackWindow time
+singleton time = RollbackWindowC time time
+
+rollForward
+    :: Ord time
+    => time
+    -> RollbackWindow time
+    -> Maybe (RollbackWindow time)
+rollForward newTip w =
+    if tip w <= newTip
+        then Just (RollbackWindowC (finality w) newTip)
+        else Nothing
+
+data MaybeRollback a
+    = Past
+    | Present a
+    | Future
+
+rollBackward
+    :: Ord time
+    => time
+    -> RollbackWindow time
+    -> MaybeRollback (RollbackWindow time)
+rollBackward newTip w =
+    if tip w < newTip
+        then Future
+        else
+            if finality w <= newTip
```

Comment:
same as above

Reply:
After discussion: `tip w < newTip` is more agreeable with `member`.